### PR TITLE
Add spatial functions

### PR DESCRIPTION
**Pull request by lossyrob**
This PR adds support for the `ST_DISTANCE`, `ST_WITHIN` and `ST_INTERSECTS` spatial functions, utilizing geospatial functionality in [turf.js](https://turfjs.org/).

The `ST_INTERSECTS` and `ST_WITHIN` functionality should match Cosmos DB's implementation.

The `ST_DISTANCE` method will match Cosmos DB's for measuring distance between points. However, due to limitations of `turf.js`, distance between geometries other than points are measured from their centroid. If others need more accurate distance measurements between non-point geometries, then that work can be accomplished in a follow up issue.

There are two other spatial functions that are unimplemented - `ST_ISVALID` and `ST_ISVALIDDETAIL`. These two methods are for validating GeoJSON and do not seem as useful for development servers. One could use [geojson](https://github.com/mapbox/geojsonhint) or `jsonschema` validation to accomplish this, but it would introduce even more complexity and dependencies, and I'm not clear on the value of those methods for the development server.

### CODE REVIEW

File: src/builtin-functions.ts [hunk 1]
```diff
@@ -275,7 +276,7 @@ export const TOSTRING = def(
       return undefined;
     }
     if (t === "object" || t === "array") {
-      JSON.stringify(v);
+      return JSON.stringify(v);
```
Comment (lossyrob) [inline src/builtin-functions.ts:279]: Saw this wasn't returning a value; added an unrelated commit to this PR to add the return statement.

Comment (nkzawa) [reply src/builtin-functions.ts:279]: good catch 🙏

**Review by nkzawa — APPROVED** [review]
Looks awesome, thank you!
